src/signalrclient/messagepack_hub_protocol.cpp

```diff
@@ -153,20 +153,20 @@ namespace signalr
         case signalr::value_type::map:
         {
             const auto& obj = v.as_map();
-            packer.pack_map((uint32_t)obj.size());
+            packer.pack_map(static_cast<uint32_t>(obj.size()));
             for (auto& val : obj)
             {
-                packer.pack_str((uint32_t)val.first.size());
-                packer.pack_str_body(val.first.data(), (uint32_t)val.first.size());
+                packer.pack_str(static_cast<uint32_t>(val.first.size()));
+                packer.pack_str_body(val.first.data(), static_cast<uint32_t>(val.first.size()));
                 pack_messagepack(val.second, packer);
             }
             return;
         }
         case signalr::value_type::binary:
         {
             const auto& bin = v.as_binary();
-            packer.pack_bin((uint32_t)bin.size());
-            packer.pack_bin_body((char*)bin.data(), (uint32_t)bin.size());
+            packer.pack_bin(static_cast<uint32_t>(bin.size()));
+            packer.pack_bin_body(reinterpret_cast<const char*>(bin.data()), static_cast<uint32_t>(bin.size()));
```

Comment:
Example: The C cast here was casting away the constness of the pointer. But the `reinterpret_cast` has a compile time error if you try to do that.